src/mixins/createFormLabelMixin.ts

```diff
@@ -0,0 +1,164 @@
+import { VNodeProperties } from '@dojo/interfaces/vdom';
+import compose, { ComposeFactory } from '@dojo/compose/compose';
+import { assign } from '@dojo/core/lang';
+import { NodeAttributeFunction, DNode, Widget, WidgetOptions, WidgetState, WidgetProperties, FormLabelProperties } from './../interfaces';
+import { v } from '../d';
+
+export interface FormLabelMixinProperties extends WidgetProperties {
+	/**
+	 * The form widget's name
+	 */
+	name?: string;
+
+	/**
+	 * The current value
+	 */
+	value?: string;
+
+	/**
+	 * The type of the form field (equates to the `type` attribute in the DOM)
+	 */
+	type?: string;
+
+	/**
+	 * Prevents the user from interacting with the form field
+	 */
+	disabled?: boolean;
+
+	/**
+	 * Label settings for form label text, position, and visibility
+	 */
+	label?: string | FormLabelProperties;
+
+	/**
+	 * Accessibility attributes
+	 */
+	checked?: boolean;
+	descriptionID?: string;
+	inputmode?: string;
```

Comment:
I left inputmode lowercase to keep it the same as the HTML attribute. I suppose we could put 'aria-describedby' in quotes and make all of them correspond directly to HTML attributes, if you think that's more consistent.